app/src/test/java/com/architectcoders/aacboard/datasource/remote/RemoteDataSourceImplTest.kt

```diff
@@ -0,0 +1,98 @@
+package com.architectcoders.aacboard.datasource.remote
+
+import com.architectcoders.aacboard.data.datasource.remote.RemoteDataSource
+import com.architectcoders.aacboard.domain.data.Error
+import com.architectcoders.aacboard.domain.data.Error.NoMatchFound
+import com.architectcoders.aacboard.domain.data.Response
+import com.architectcoders.aacboard.domain.data.cell.CellPictogram
+import com.architectcoders.aacboard.network.ArasaacNetworkInstance
+import com.architectcoders.aacboard.network.service.ArasaacService
+import com.architectcoders.aacboard.testrules.CoroutinesTestRule
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.runTest
+import okhttp3.mockwebserver.MockResponse
+import okhttp3.mockwebserver.MockWebServer
+import org.junit.After
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import java.net.HttpURLConnection
+
+@OptIn(ExperimentalCoroutinesApi::class)
+class RemoteDataSourceImplTest {
+
+    @get:Rule
+    val coroutinesTestRule = CoroutinesTestRule()
+
+    lateinit var arasaacService: ArasaacService
+    lateinit var remoteDataSource: RemoteDataSource
+
+    lateinit var server: MockWebServer
+
+    @Before
+    fun setup() {
+        arasaacService =
+            ArasaacNetworkInstance.createArasaacNetworkInstance("http://localhost:8080")
+        remoteDataSource = RemoteDataSourceImpl(arasaacService)
+        server = MockWebServer()
+        server.start(8080)
+    }
+
+    @After
+    fun tearDown() {
+        server.shutdown()
+    }
+
+
+    @Test
+    fun `no pictogram found, http error 404, NoMathFound error returned`(): Unit = runTest {
```

Comment:
We haven't talk about this in the group but we could follow the GIVEN WHEN THEN approach for the tests naming if you like or not. Opened for discussion.

Reply:
Updated in #57 trying to fit test method's names within detekt maximum line length